rero_ils/theme/menus.py

```diff
@@ -321,8 +329,14 @@ def init_menu_profile():
     item = current_menu.submenu('main.profile.edit_profile')
     rero_register(
         item,
-        endpoint='invenio_userprofiles.profile',
-        visible_when=lambda: current_user.is_authenticated,
+        endpoint='users.profile',
+        endpoint_arguments_constructor=lambda: dict(
+            viewcode=request.view_args.get(
+                'viewcode', current_app.config.get(
+                    'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'))),
+        visible_when=lambda: not current_app.config.get(
+            'RERO_PUBLIC_USERPROFILES_READONLY', False) and
+        current_user.is_authenticated,
```

Comment:
This lambda function is used three times. You should create a function and use this function as `visible_when` argument.